arch/arm/src/max326xx/max32690/max32690_gpio.c

```diff
@@ -0,0 +1,448 @@
+ * Private Data
+ ****************************************************************************/
+
+spinlock_t g_max32690_gpio_lock = SP_UNLOCKED;
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: max32690_default
+ *
+ * Description:
+ *   Configure a pin as a default GPIO input as it was on reset
+ *
+ ****************************************************************************/
+
+static void max32690_default(max32690_gpio_regs_t *gpio,
+                             max32690_pinconfig_t *conf)
+{
+    uint32_t pinmask = 1 << conf->pin;
+
+      /* Disable interrupts and wake-up events */
+
+    gpio->inten_clr =  pinmask;  /* Pin interrupt disabled triggered */
+    gpio->intmode  &= ~pinmask;  /* Level triggered */
+    gpio->intpol   &= ~pinmask;  /* Input low triggers */
+    gpio->dualedge &= ~pinmask;  /* Disable dual edge */
+    gpio->wken_clr  =  pinmask;  /* Disable wakeup */
+
+      /* Make the pins an input an clear output value */
+
+    gpio->outen_clr =  pinmask;  /* Disable output drivers */
+    gpio->out_clr   =  pinmask;  /* Set the output value to zero */
+
+      /* Set alternate functions to I/O */
+
+    gpio->en0_set   =  pinmask;
+    gpio->en1_clr   =  pinmask;
+    gpio->en2_clr   =  pinmask;
+
+      /* enable input connection to pin */
+
+    gpio->inen     |= pinmask;
+
+      /* Reset drive strength */
+
+    gpio->ds0      &= ~pinmask;
+    gpio->ds1      &= ~pinmask;
+
+      /* Disable pull up and pull down */
+
+    gpio->padctrl0 &= ~pinmask;
+    gpio->ps       &= ~pinmask;
+
+      /* Set VVoltage to V_DDIOH */
+
+    gpio->vssel |= pinmask;
+
+      /* Disable slew and hysteresis */
+
+    gpio->hysen    &= ~pinmask;
+    gpio->srsel    &= ~pinmask;
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: max32690_get_gpio_bank_regptr
+ *
+ * Description:
+ *   Get the GPIO access struct address based in the GPIO Bank
+ *
+ ****************************************************************************/
+
+max32690_gpio_regs_t *max32690_get_gpio_bank_regptr(int bank)
+{
+    max32690_gpio_regs_t *gpio_regs = 0;
+
+    switch (bank)
+    {
+        case 0: gpio_regs = (max32690_gpio_regs_t *)MAX326_GPIO0_BASE;
+            break;
+        case 1: gpio_regs = (max32690_gpio_regs_t *)MAX326_GPIO1_BASE;
+            break;
+        case 2: gpio_regs = (max32690_gpio_regs_t *)MAX326_GPIO2_BASE;
+            break;
+        case 3: gpio_regs = (max32690_gpio_regs_t *)MAX326_GPIO3_BASE;
+            break;
```

Comment:
I don't know why nxstyle script didn't get it, but "case X:" should be the only thing in a line.

Refer to max32660/max32660_lowputc.c line 334